### PR TITLE
teuchos(cleaning): remove old `remove_const_t` trait

### PR DESCRIPTION
**Pull request by romintomasetti**
@trilinos/teuchos

## Motivation

Removing old in-house version of `std::remove_const_t`.

@bartlettroscoe

### CODE REVIEW

File: packages/teuchos/core/src/Teuchos_TypeNameTraits.hpp [hunk 1]
```diff
@@ -64,6 +62,8 @@
 
 #include <typeinfo>
 
+#include "Teuchos_ConfigDefs.hpp"
```
Comment (romintomasetti) [inline packages/teuchos/core/src/Teuchos_TypeNameTraits.hpp:65]: Note: Needed to get `TEUCHOSCORE_LIB_DLL_EXPORT.

**Review by bartlettroscoe — APPROVED** [review]
Looks good to me.  And with Trilinos 15.0 to be released next, this is a good time to remove backward compatibility (in case Trilinos customers were using this traits class).

**Comment by maartenarnst** [comment]
Hi @bartlettroscoe, would you have a moment to add the `AT: PRE-TEST INSPECTED` label? @romintomasetti and I don't have the permissions to set it. Thanks in advance!

**Comment by bartlettroscoe** [comment]
> Hi @bartlettroscoe, would you have a moment to add the `AT: PRE-TEST INSPECTED` label? @romintomasetti and I don't have the permissions to set it. Thanks in advance!

@romintomasetti, done.  I forget that approving the PR does not let the autotester run PR builds.

**Comment by bartlettroscoe** [comment]
CC: @trilinos/nox 

One test [NOX_brusselator_MPI_4](https://trilinos-cdash.sandia.gov/testSummary.php?project=1&name=NOX_brusselator_MPI_4&date=2023-09-08) failed due to jumbled output showing:

```
Test passe	Time to fill Jacobian [3] --> 0.00235081 sec.

	Time to fill Jacobian [3] --> 0.00236106 sec.

	Time to fill Jacobian [1] --> 0.00239301 sec.

	Time to fill Jacobian [1] --> 0.00235081 sec.

	Time to fill Jacobian [1] --> 0.00236106 sec.

	Time to fill Jacobian [2] --> 0.00239205 sec.

	Time to fill Jacobian [2] --> 0.00235081 sec.

	Time to fill Jacobian [2] --> 0.00236106 sec.

d!
```

due to:

`Required regular expression not found. Regex=[Test passed!<br /> ]`

I wonder how many tests in Trilinos are susceptible to race conditions in printing?

Setting `AT: RETEST` and crossing fingers

**Comment by bartlettroscoe** [comment]
> Picking one or the other is OK - though cout is "template friendly", unlike printf - but mixing them is probably not a good idea

@rppawlo, I think that is okay as long as you know what is happening and you flush the streams before printing important lines like this.  But it is just error prone to only print to std::cout so you don't need to worry about buffering and flushing.

**Comment by bartlettroscoe** [comment]
> Picking one or the other is OK - though cout is "template friendly", unlike printf - but mixing them is probably not a good idea

@rppawlo, Several of these have been reported over the years (mostly by me, as shown [here](https://github.com/trilinos/Trilinos/issues?q=is%3Aissue+jumbled)).  I am wondering how many for these types of tests are lingering in the Trilinos test suite waiting to bring down PR builds?  (I would guess dozens more at least.  I think most people just put `AT: RETEST` on the PRs and never even look at the failures on CDash.)

**Pull request merged**